gsl_programmation/models.py

```diff
@@ -55,58 +37,36 @@ class Meta:
                 fields=(
                     "annee",
                     "type",
-                    "perimetre_region",
-                    "perimetre_departement",
-                    "perimetre_arrondissement",
-                ),
-                nulls_distinct=False,  # important because "perimetre_*" fields are nullable
-            ),
-            models.CheckConstraint(
-                name="only_one_perimeter",
-                violation_error_message="Un seul type de périmètre doit être renseigné parmi les trois possibles.",
-                condition=Q(perimetre_region__isnull=False)
-                ^ Q(perimetre_departement__isnull=False)
-                ^ Q(perimetre_arrondissement__isnull=False),
-            ),
-            models.CheckConstraint(
-                condition=~Q(type="DSIL")
-                | (Q(deleguee_by__isnull=True) & Q(perimetre_region__isnull=False))
-                | (
-                    Q(deleguee_by__isnull=False)
-                    & (
-                        Q(perimetre_departement__isnull=False)
-                        ^ Q(perimetre_arrondissement__isnull=False)
-                    )
```

Comment:
Petit instant 1jour1leçon (merci @mjeammet pour cette traduction parfaite de _today I learned_).

Au début j'avais écrit des contraintes du genre `perimetre__departement__isnull=False` mais ça marche pas.

On se retrouve avec des erreurs de ce type :

> (models.E041) 'constraints' refers to the joined field _truc truc_

Ça dit en gros qu'on ne peut pas assurer la contrainte côté base de données car ça colle à plusieurs tables à la fois ; on doit assurer la contrainte en Python dans la méthode `clean()`.

Source [stackoverflow](https://stackoverflow.com/questions/75640162/django-showing-error-constraints-refers-to-the-joined-field) comme d'hab